Inline JSDoc parameter type info for arrow-functions is mistaken for JSDoc documentation on the function

eslint 2.0

Because WebStorm is able to use the type information I use inline JSDoc type info (which is valid JSDoc). Such type-info of course is just that - it has nothing to do with documenting a function (@param,@returns) - it's just type information.

Second, I use inline arrow functions and I leave out the parenthesis when there is only a single parameter to such an inline arrow function.

Example:

```
// Without () around the single parameter: eslint problem!
Promise.resolve(42)
.then(/**number*/ n => n * 2);

// With () around the parameter "n"
Promise.resolve(42)
.then((/**number*/ n) => n * 2);
```

Problem: In the forst example eslint assumes that I'm trying to document a function and reports 

     error  Missing JSDoc @returns for function                      valid-jsdoc

![Screenshot](https://cloud.githubusercontent.com/assets/199179/13026894/7df7da4e-d239-11e5-99d7-6c722d39d50e.png)

I experimented and slept over it a bit more. I have `require-jsdoc` turned OFF - so any errors are only due to rule `valid-jsdoc`.

Two observations:

1) A simple type-only JSDoc coment *is* valid JSDoc.

2) I have in my `valid-jsdoc` rule

    "requireReturn": false,

yet I get errors like

    166:7   error  Missing JSDoc @returns for function                      valid-jsdoc


So as I see it, the rule is called "valid JSDoc". My JSDoc *is* valid. It should not complain. I have *not* enabled the options that "require return".

--- a/docs/rules/valid-jsdoc.md
+++ b/docs/rules/valid-jsdoc.md
@@ -31,6 +31,8 @@ This rule enforces valid and consistent JSDoc comments. It reports any of the fo
 
 This rule does not report missing JSDoc comments for classes, functions, or methods.
 
+**Note:** This rule does not support all of the Google Closure documentation tool's use cases. As such, some code such as `(/**number*/ n => n * 2);` will be flagged as missing appropriate function JSDoc comments even though `/**number*/` is intended to be a type hint and not a documentation block for the function. We don't recommend using this rule if you use type hints in this way.
+
 Examples of **incorrect** code for this rule:
 
 ```js

